packages/server/graphql/mutations/helpers/notifications/Notifier.ts

```diff
@@ -0,0 +1,45 @@
+import {DataLoaderWorker} from '../../../graphql'
+
+/**
+ * Empty Notifier prototype to allow iterating the methods typesafe
+ */
+export const EmptyNotifier = {
```

Comment:
-1 this pattern of abstract noop methods isn't very conventional. My concern is if other devs start copying this pattern, would the codebase get harder to read?

As constructed, `IntegrationNotifier` is essentially a singleton, it's just built without using the `class` syntactical sugar. 

2 options that are a little more expressive:

```ts
class NotifierManager {
  // put each type in as a method
  endMeeting(dataLoader, meetingId, teamId) {
    return Promise.allSettled(notifiers.map((notifier.endMeeting(dataLoader, meetingId, teamId)
  } 
}
export IntegrationManager = new NotifierManager()
```

```ts
interface Notifier {
  endMeeting: (meetingId: string) => Promise<NotifyResponse>
  startMeeting: (meetingId: string) => Promise<NotifyResponse>
}

interface NotifierManager extends Notifier {}
class NotifierManager {
  constructor() {
    // use a helper to do the work
    const methods = ['endMeeting', 'startMeeting']
    methods.forEach((method) => {
      this[method] = async (...args: any[]) => {
        return notifiers.map((notifier) => notifier[method](...args))
      }
    })
  }
}

export default IntegrationNotifier = new NotifierManager()
```

Reply:
Going with the copy & paste method as there is no way to ensure all methods are listed. That's why I had introduced the empty object placeholder to have the type information available at runtime.